### PR TITLE
[core] Global Clock Error

### PR DESCRIPTION
**Pull request by Nicolas-Vanz**

### CODE REVIEW

**Comment by joaovicentesouto** [comment]
This PR broke upside layers, we must do rebase and verify if still failing or it's running from HAL until multikernel.

File: include/nanvix/hal/cluster/timer.h [hunk 1]
```diff
@@ -80,6 +80,8 @@
 	 */
 	EXTERN void timer_init(unsigned freq);
 
+	EXTERN uint64_t clock_get_error(void);
```
Comment (ppenna) [inline include/nanvix/hal/cluster/timer.h:83]: Could add a documentation please? =D

File: src/hal/cluster/timer.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include <nanvix/hal/cluster/timer.h>
+
+PRIVATE uint64_t clock_error = 0ULL;
+
+PUBLIC uint64_t clock_get_error(void)
+{
+	return clock_error;
+}
+
+PUBLIC void timer_init(unsigned freq)
```
Comment (ppenna) [inline src/hal/cluster/timer.c:34]: Could add a documentation please? =D

File: src/hal/cluster/timer.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <nanvix/hal/cluster/timer.h>
+
+PRIVATE uint64_t clock_error = 0ULL;
+
+PUBLIC uint64_t clock_get_error(void)
```
Comment (ppenna) [inline src/hal/cluster/timer.c:29]: Could add a documentation please? =D

File: src/hal/cluster/timer.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <nanvix/hal/cluster/timer.h>
+
+PRIVATE uint64_t clock_error = 0ULL;
```
Comment (ppenna) [inline src/hal/cluster/timer.c:27]: Could add a documentation please? =D

**Review by ppenna — COMMENTED** [review]
Why do we need to update contrib/barelib?

**Review by ppenna — DISMISSED** [review]

**Review by ppenna — APPROVED** [review]

**Review by ppenna — APPROVED** [review]

**Pull request merged**